orb-ui/src/engine/center/alert.rs

```diff
@@ -53,25 +67,49 @@ impl<const N: usize> Animation for Alert<N> {
         idle: bool,
     ) -> AnimationState {
         let mut time_acc = 0.0;
+        let mut color = Argb::OFF;
 
         // sum up each edge duration
         for (i, &time) in self.pattern.iter().enumerate() {
             time_acc += time;
             // make sure we use the color associated with the local animation time
-            if self.phase < time_acc {
-                // first edge depends on `start_on`
+            let smooth = if let Some(s) = self.smooth_transitions.as_ref() {
+                s[i]
+            } else {
+                0.0
+            };
+
+            if self.phase < time_acc + smooth {
+                // first edge [i = 0] depends on `start_on`
                 let mod_res = usize::from(self.start_on);
-                self.current_color = if i % 2 == mod_res {
+                self.current_solid_color = if i % 2 == (1 - mod_res) {
                     self.target_color
                 } else {
                     Argb::OFF
                 };
+                color = self.current_solid_color;
+                break;
+            } else if smooth != 0.0
+                && self.phase < time_acc + smooth + self.pattern[i + 1]
+                && self.phase >= time_acc + smooth
+            {
+                // transition between edges
+                let period = self.pattern[i + 1] - smooth;
+                let t = self.phase - time_acc - smooth;
+                let intensity = if self.current_solid_color == Argb::OFF {
+                    // starts at intensity 0
+                    1.0 - (t * (PI / 2.0) / period).cos()
+                } else {
+                    // starts at intensity 1
+                    (t * (PI / 2.0) / period).cos()
+                };
+                color = self.target_color * intensity;
```

Comment:
@fouge can you add a unit test for some of this math? To just check a few values that you expect. This would give me more confidence in the math. Also, giving this a function name with a couple tests may make it easier for me to understand what its doing.